### ISSUE TITLE
Fix FP S1131(`no-trailing-spaces`): Markdown code comments

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/SonarJS#1282]
**I want to report a bug.**

SonarJS raises a rule violation for `javascript:TrailingWhitespace` also in code comment blocks.
Markdown-based comment support is quite common with big-name documentation engines all offering varying levels of support for it. In Markdown a pair of trailing spaces has actual syntactic meaning: it signals a hard newline.

**SonarJS version:** 5.0 (build 6962)
**Rule key:** javascript:TrailingWhitespace

**Reproducer** 
```js
/**
 * This is a code comment that will be processed as Markdown  
 * The line above is terminated with two trailing spaces to create a hard newline.
 * Sadly it will cause SonarJS to raise a `javascript:TrailingWhitespace` violation.
 */
function someFunctionToDocument() {}
```

**Expected behavior**
SonarJS should not raise the rule violation when whitespace can have syntactic meaning, as it does in Markdown.

**Comment by ilia-kebets-sonarsource** [comment]
This code isn't flagged anymore. Maybe it is because of how we analyze commented code now.

**Comment by yassin-kammoun-sonarsource** [comment]
I think it has nothing to do with the way we analyze commented code. The issue was raised by `no-trailing-spaces`, which is an ESLint rule we just enable. We can close the issue since it's no longer reported, but it would be nice to understand why.
